app/src/main/java/io/homeassistant/companion/android/sensors/CarSensorManager.kt

```diff
@@ -123,10 +166,23 @@ class CarSensorManager :
         )
     }
 
+    private lateinit var context: Context
+
+    private val isAutomotive get() = context.packageManager.hasSystemFeature(PackageManager.FEATURE_AUTOMOTIVE)
+
+    private val carSensorsList get() = allSensorsList.filter { (isAutomotive && it.automotiveEnabled) || (!isAutomotive && it.autoEnabled) }
+    private val sensorsList get() = carSensorsList.map { it.sensor }
+
+    private fun allDisabled(): Boolean = sensorsList.none { isEnabled(context, it) }
+
+    private fun connected(): Boolean = HaCarAppService.carInfo != null
+
     override val name: Int
         get() = R.string.sensor_name_car
 
     override suspend fun getAvailableSensors(context: Context): List<SensorManager.BasicSensor> {
+        this.context = context.applicationContext
```

Comment:
Those lateinit context set are a bit sketchy. I had to set it from more methods, as it wasn't set in some calls. I added it to every method accepting a context, just in case it happens again in the future.

I'm considering 2 other options: Adding it on construction (Not sure if possible), or simply drilling it through method calls, so we're sure we actually have a context when calling something that needs it.

I'd like to see more opinions about this

Reply:
So we do actually use a `latestContext` method of storing the latest context received in the app for some sensors, very similar to what you are doing here.

https://github.com/home-assistant/android/blob/master/common/src/main/java/io/homeassistant/companion/android/common/sensors/LightSensorManager.kt#L57

Reply:
I see. Renaming to latestContext, to keep the same naming as the other sensors